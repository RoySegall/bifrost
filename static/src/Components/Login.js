import React from 'react';
import axios from 'axios';
import FormError from './FormError';


export default class Login extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      username: null,
      password: null,
      errors: {
        type: null,
        message: '',
      },
    };
  }

  setError(error) {
    this.setState({errors: {type: 'error', message: error}});
  }

  setSuccess(message) {
    this.setState({errors: {type: 'success', message: message}});
  }

  resetError() {
    this.setState({errors: {type: null, message: null}});
  }

  login(event) {
    event.preventDefault();
    this.resetError();

    if (!this.state.username) {
      this.setError('Username is required');
      return;
    }

    if (!this.state.password) {
      this.setError('Password is required');
      return;
    }

    axios.post(`${process.env.REACT_APP_BACKEND}/auth/login`, {
      username: this.state.username,
      password: this.state.password,
    })
      .then(resp => {
        this.setSuccess('You are now logged!');
        localStorage.setItem('token', resp.data.token);
        setTimeout(() => {
          window.location.href = '/';
        }, 1500);
      })
      .catch(error => {
        this.setError(`Something went wrong: ${error.response.data.error}`);
      })
  }

  username() {
    return <div className="md:flex md:items-center mb-6">
      <div className="md:w-1/3">
        <label className="block text-black font-bold md:text-right mb-1 md:mb-0 pr-4" htmlFor="inline-full-name">
          Username
        </label>
      </div>
      <div className="md:w-2/3">
        <input
          onChange={(event) => {
            this.setState({username: event.currentTarget.value})
          }}
          id="username"

          aria-describedby="emailHelp"
          placeholder="Enter username"
          className="
                bg-gray-200 appearance-none
                border-2 border-gray-200 rounded
                w-full py-2 px-4
                text-black leading-tight
                focus:outline-none focus:bg-white focus:border-purple-500

          " type="text" />
      </div>
    </div>
  }

  password() {
    return <div className="md:flex md:items-center mb-6">
      <div className="md:w-1/3">
        <label className="block text-black font-bold md:text-right mb-1 md:mb-0 pr-4" htmlFor="inline-username">
          Password
        </label>
      </div>
      <div className="md:w-2/3">
        <input
          type="password"
          id="password"
          placeholder="Enter password"
          onChange={(event) => {
            this.setState({password: event.currentTarget.value})
          }}
          className="
            bg-gray-200 appearance-none border-2 border-gray-200
            rounded w-full py-2 px-4 text-gray-700 leading-tight
            focus:outline-none focus:bg-white focus:border-purple-500
          " />
      </div>
    </div>
  }

  submitButton() {
    return <div className="md:flex md:items-center">
      <div className="md:w-1/3"></div>
      <div className="md:w-2/3">
        <button className="button" type="button"onClick={(event) => {
          this.login(event)
        }}>Sign Up</button>
      </div>
    </div>
  }

  render() {
    return <div>
      <FormError type={this.state.errors.type} message={this.state.errors.message}/>

      <form className="w-full max-w-sm p-4 m-auto m-0">
        {this.username()}
        {this.password()}
        {this.submitButton()}
      </form>
    </div>
  }

}
