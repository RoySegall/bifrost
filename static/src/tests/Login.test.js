import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../Components/Login'

test('Testing the login component', () => {

    const component = renderer.create(<Login />);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});