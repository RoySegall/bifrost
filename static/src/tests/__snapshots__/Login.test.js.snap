// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the login component 1`] = `
<div>
  <div
    className="flex flex-row justify-between mt-2 mb-2"
  >
    <form
      className="margin-0-auto w-6/12"
    >
      <div
        className="flex-row pb-2"
      >
        <label
          className="align-bottom mr-5 text-black font-bold"
          htmlFor="username"
        >
          Username
        </label>
        <input
          className="align-bottom m-4 p-2 ml-5 mb-0 pb-0 border-blue-500 border-b focus:bg-blue-100 w-3/4"
          id="username"
          onChange={[Function]}
          placeholder="Username"
          type="text"
        />
      </div>
      <div
        className="flex-row pb-2"
      >
        <label
          className="align-bottom mr-5 text-black font-bold"
          htmlFor="username"
        >
          Password
        </label>
        <input
          className="align-bottom m-4 p-2 ml-5 mb-0 pb-0 border-blue-500 border-b focus:bg-blue-100 w-3/4"
          id="password"
          onChange={[Function]}
          placeholder="Enter password"
          type="password"
        />
      </div>
      <div
        className="flex-row text-center"
      >
        <button
          className="mt-4 button"
          onClick={[Function]}
        >
          Submit
        </button>
      </div>
    </form>
  </div>
</div>
`;
